app/interactors/create_order.rb

```diff
@@ -0,0 +1,44 @@
+class CreateOrder
+  include Interactor
+
+  def call
+    @order = context.user.orders.new(context.order_params)
+    subtotal = context.user.cart.total_price
+    tax = subtotal * 0.1
+    total = subtotal + tax
+    payment_method = "COD: Cash On Delivery"
+
+    @order.update(subtotal: subtotal, tax: tax, total: total, payment_method: payment_method)
+
+    if @order.save
+      create_order_items
+      update_cart_totals
```

Comment:
Do you need a transaction?

Reply:
yes over here i'm managing the items in the cart but not using a transaction block since due to time constraint. If you could give me more time now then i can make the code according to all the proper rules and regulations. Has to take the quick way because of 1hr time constraint.